src/entry.cc

```diff
@@ -1,128 +1,206 @@
-  delete (pST);
+void fts5_simple_xDelete(Fts5Tokenizer* p) {
+    auto* pST = (simple_tokenizer::SimpleTokenizer*)p;
+    delete (pST);
 }
 
 /*
 ** Return a pointer to the fts5_api pointer for database connection db.
 ** If an error occurs, return NULL and leave an error in the database
 ** handle (accessible using sqlite3_errcode()/errmsg()).
 */
-static int fts5_api_from_db(sqlite3 *db, fts5_api **ppApi) {
-  sqlite3_stmt *pStmt = 0;
-  int rc;
-
-  *ppApi = 0;
-  rc = sqlite3_prepare(db, "SELECT fts5(?1)", -1, &pStmt, 0);
-  if (rc == SQLITE_OK) {
-    sqlite3_bind_pointer(pStmt, 1, reinterpret_cast<void *>(ppApi), "fts5_api_ptr", 0);
-    (void)sqlite3_step(pStmt);
-    rc = sqlite3_finalize(pStmt);
-  }
-
-  return rc;
+static int fts5_api_from_db(sqlite3* db, fts5_api** ppApi) {
+    sqlite3_stmt* pStmt = 0;
+    int rc;
+
+    *ppApi = 0;
+    rc = sqlite3_prepare(db, "SELECT fts5(?1)", -1, &pStmt, 0);
+    if (rc == SQLITE_OK) {
+        sqlite3_bind_pointer(pStmt, 1, reinterpret_cast<void*>(ppApi),
+                             "fts5_api_ptr", 0);
+        (void)sqlite3_step(pStmt);
+        rc = sqlite3_finalize(pStmt);
+    }
+
+    return rc;
 }
 
 #ifdef USE_JIEBA
-static void jieba_dict(sqlite3_context *pCtx, int nVal, sqlite3_value **apVal) {
-  if (nVal >= 1) {
-    const char *text = (const char *)sqlite3_value_text(apVal[0]);
-    if (text) {
-      std::string tmp(text);
-      char sep = '/';
+static void jieba_dict(sqlite3_context* pCtx, int nVal, sqlite3_value** apVal) {
+    if (nVal >= 1) {
+        const char* text = (const char*)sqlite3_value_text(apVal[0]);
+        if (text) {
+            std::string tmp(text);
+            char sep = '/';
 #ifdef _WIN32
-      sep = '\\';
+            sep = '\\';
 #endif
-      if (tmp.back() != sep) {  // Need to add a
-        tmp += sep;             // path separator
-      }
-      simple_tokenizer::jieba_dict_path = tmp;
-      sqlite3_result_text(pCtx, tmp.c_str(), -1, SQLITE_TRANSIENT);
-      return;
+            if (tmp.back() != sep) {  // Need to add a
+                tmp += sep;           // path separator
+            }
+            simple_tokenizer::jieba_dict_path = tmp;
+            sqlite3_result_text(pCtx, tmp.c_str(), -1, SQLITE_TRANSIENT);
+            return;
+        }
     }
-  }
-  sqlite3_result_null(pCtx);
+    sqlite3_result_null(pCtx);
 }
 
-static void jieba_query(sqlite3_context *pCtx, int nVal, sqlite3_value **apVal) {
-  if (nVal >= 1) {
-    const char *text = (const char *)sqlite3_value_text(apVal[0]);
-    if (text) {
-      int flags = 1;
-      if (nVal >= 2) {
-        flags = atoi((const char *)sqlite3_value_text(apVal[1]));
-      }
-      std::string result = simple_tokenizer::SimpleTokenizer::tokenize_jieba_query(text, (int)std::strlen(text), flags);
-      sqlite3_result_text(pCtx, result.c_str(), -1, SQLITE_TRANSIENT);
-      return;
+static void query(sqlite3_context* pCtx, int nVal, sqlite3_value** apVal) {
```

Comment:
You need to register this using sqlite3_create_function? And can we re-use jieba_query to achieve the same goal? Like what we did in simple_query,  `if (nVal >= 2) ` we parse extra options.

Because you format code, it's hard for me to review the logic change